src/lib/config/ModelConfig.cc

```diff
@@ -886,6 +886,29 @@ void ModelConfig::postValidation() {
   // Record any unlinked port names
   for (const auto& prt : portnames)
     invalid_ << "\t- " << prt << " has no associated reservation station\n";
+
+  // Ensure the L1-[Data|Instruction]-Memory:Interface-Type restrictions are
+  // enforced
+  // Currently, only outoforder core types can use non-Flat L1-Data-Memory
+  // interfaces
+  if (configTree_["Core"]["Simulation-Mode"].as<std::string>() !=
+      "outoforder") {
+    std::string l1dType =
+        configTree_["L1-Data-Memory"]["Interface-Type"].as<std::string>();
+    if (l1dType != "Flat")
+      invalid_
+          << "\t- Only a 'Flat' L1-Data-Memory Interface-Type may be used "
+             "for a non-outoforder Simulation-Mode. Interface-Type used is "
```

Comment:
Rather than "a non-outoforder", have the message print out the simulation mode chosen.

Reply:
Good idea